### PR TITLE
[Magento2] Add multisite configuration options & Small fixes

### PR DESCRIPTION
**Pull request by jeliebig**
This PR makes a few changes to the existing Magento2 app.

## General fixes

- Magento 2.3 now creates projects with the correct version again.
- Magento 2.3 and 2.4 can be installed correctly again by changing the db image to [mariadb](https://hub.docker.com/_/mariadb/) 10.6 (which is the latest version supported by Magento at the time).
- Additionally Magento 2.4 requires a newer php version, so the image version was updated to 8.1-fpm as well.

## Multisite configuration

Magento is able to [host multiple websites and stores](https://experienceleague.adobe.com/docs/commerce-operations/configuration-guide/multi-sites/ms-nginx.html), but until now there was no easy way to access them using riptide.

By replacing the main entrypoint of `nginx.conf.sample` with a slightly customized one and adding a few options to the default nginx configuration `magento-nginx.conf`, you are now able to use riptide's project configuration to add multiple websites/stores for Magento and expose them on different subdomains.

### Example

<details>
<summary>Magento 2.4 riptide config</summary>

```yml
project:
  name: magento-demo-2-4
  src: src
  app:
    $ref: /app/magento2/ce/2.4
    services:
      www:
        environment:
          SUBDOMAIN_TEST1_CODE: test
          SUBDOMAIN_TEST2_CODE: test2
          SUBDOMAIN_TEST3_CODE: test3
      varnish:
        additional_subdomains:
          - test1
          - test2
          - test3
```
</details> 

### Configuration

1. Add the new subdomains to the service with the main role (in this case `varnish`) using the key `additional_subdomains`.
2. Add the `MAGE_RUN_CODE` for every website as an environment variable to the `www` service using the following key: `SUBDOMAIN_{ additional subdomain (uppercase) }_CODE`.
3. If you want to change the `MAGE_RUN_TYPE` you can do this by setting the environment variable (also named: `MAGE_RUN_TYPE`) to a different value for the `www` service. (Default: `website`)

---

Depends on: theCapypara/riptide-lib#45

### CODE REVIEW

File: app/magento2/base.yml [hunk 1]
```diff
@@ -132,6 +147,7 @@ app:
 
     db:
       $ref: /service/mysql/5.7
+      image: mariadb:10.6
```
Comment (theCapypara) [inline app/magento2/base.yml:150]: Why did you add this, this seems unrelated?

File: app/magento2/ce/2.4.yml [hunk 2]
```diff
@@ -51,12 +51,21 @@ app:
...
+        which is done by setting an environment variable for each subdomain you added using the following key: SUBDOMAIN_<uppercase subdomain name>_CODE
+  
+        You can find an example configuration here: https://github.com/theCapypara/riptide-repo/pull/28#Example
 
       You can change the settings in step 3 to your likings, see the installation guide at
         https://devdocs.magento.com/guides/v2.4/install-gde/install/cli/install-cli.html
 
   services:
     php:
+      image: riptidepy/php:8.1-fpm
```
Comment (theCapypara) [inline app/magento2/ce/2.4.yml:68]: Why did you add this, this seems unrelated?

**Review by theCapypara — CHANGES_REQUESTED** [review]

**Review by theCapypara — APPROVED** [review]
Ah right, I should have read the description :)

**Review by theCapypara — APPROVED** [review]
Ah right, I should have read the description :)

**Review by theCapypara — CHANGES_REQUESTED** [review]

File: app/magento2/base.yml [hunk 3]
```diff
@@ -78,10 +86,15 @@ app:
...
       config:
+        nginx_conf_main_entry:
+          from: assets/magento-nginx-entrypoint.conf
+          to: '{{ get_working_directory() }}/nginx.conf.main_entry'
         env_php:
           from: assets/env.php
           to: '{{ get_working_directory() }}/app/etc/env.php'
       pre_start:
+        # Remove main entrypoint from nginx.conf.sample
+        - "sed -i'' -z -E 's/# PHP entry point for main application(\n|.)*}\n\n//' '{{ get_working_directory() }}/nginx.conf.sample'"
```
Comment (theCapypara) [inline app/magento2/base.yml:97]: This isn't portable, it doesn't work in CentOS 7 based images:
```
/entrypoint_riptide.sh: eval: line 219: syntax error near unexpected token `(\'\n/entrypoint_riptide.sh: eval: line 219: `exec su riptide -m -c \' /bin/sh -c "sed -i\'\' -z -E \'s/# PHP entry point for main application( |.)*} //\' \'/src/www/nginx.conf.sample\'" /usr/local/bin/docker-entrypoint.sh \'\'\n
```

Additionally this changes the versioned nginx.conf.sample in projects. We need a different approach.

**Comment by theCapypara** [comment]
@jeliebig 
I changed the nginx config mechanism:
- The Nginx Magento config file you added now contains everything that was in the sample config before + the changes for setting the store and run code
- The sample config is no longer loaded at all
- The sed command was removed since it's not needed anymore.

This is a bit of a breaking change but I think this will overall work smoother.

File: app/magento2/base.yml (hunk 1 above)
Comment (jeliebig) [reply app/magento2/base.yml:148]: Magento2 didn't work for me using the older version used by Magento1, so I only changed it here.

File: app/magento2/base.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeliebig) [reply app/magento2/base.yml]: Changing it to
```
sed -i'' -z -E 's!# PHP entry point for main application(\n|.)*}\n\n!!' '{{ get_working_directory() }}/nginx.conf.sample'
```
also doesn't make it work, right?

**Comment by jeliebig** [comment]
I see, I have no objections as long as it works! 

I originally added the `sed` command so I could avoid replacing the entire sample config in case it gets changed with an update. I assumed it was unlikely for them to change the main entrypoint part, so I thought using sed was a safe thing to do. But if it's completely fine to use the new config file for everything then it's not needed and we don't need to think about how we ensure this doesn't break.
